homework/src/index.js

```diff
@@ -1,47 +1,155 @@
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
-  }
-
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach((key) => {
-      const value = options[key];
-      if (key === 'html') {
-        elem.innerHTML = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
-    return elem;
-  }
-
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
-  }
-
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-
-  window.onload = () => main(HYF_REPOS_URL);
+function main() {
+    const url = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+
+    function createAndAppend(tagName, parent) {
+        const element = document.createElement(tagName);
+        parent.appendChild(element);
+        return element;
+    }
+
+    function fetchJSON(url) {
+        return new Promise((resolve, reject) => {
+            const xhr = new XMLHttpRequest();
+            xhr.open("GET", url, true);
+            xhr.responseType = "json";
+            xhr.onreadystatechange = () => {
+                if (xhr.readyState === 4) {
+                    if (xhr.status < 400) {
+                        resolve(xhr.response);
+                    } else {
+                        reject(new Error(`Network error : ${xhr.status} - ${xhr.statusText}`));
+                    }
+                }
+            };
+            xhr.send();
+        });
+    }
+
+
+    function renderRepositories(data) {
+
+        const root = document.getElementById('root');
+
+        const header = createAndAppend('div', root);
+        header.setAttribute('id', 'header');
+
+        const head = createAndAppend('h1', header);
+        head.innerHTML = 'HYF Repositories';
+
+        const select = createAndAppend('select', header);
+
+        const container = createAndAppend('div', root);
+        const information = createAndAppend('div', container);
+        information.setAttribute('id', 'information');
+
+        const contributors = createAndAppend('div', container);
+        contributors.setAttribute('id', 'contributors');
+
+        const contributorsList = createAndAppend('ul', contributors);
+        contributorsList.setAttribute('id', "contributors-list");
+
+
+        select.addEventListener('change', (event) => {
+            const newURL = 'https://api.github.com/repos/HackYourFuture/' + event.target.value;
```

Comment:
You should avoid putting URLs inside functions, kind of makes the code hard to reuse